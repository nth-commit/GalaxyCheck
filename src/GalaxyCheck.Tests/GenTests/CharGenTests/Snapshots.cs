using GalaxyCheck;
using Snapshooter;
using Snapshooter.Xunit;
using System.Linq;
using Xunit;

namespace Tests.V2.GenTests.CharGenTests
{
    public class Snapshots
    {
        [Fact]
        public void Values()
        {
            var seeds = Enumerable.Range(0, 3);

            foreach (var seed in seeds)
            {
                var sample = Gen.Char().Select(c => char.IsControl(c) ? "<control>" : c.ToString()).Sample(seed: seed);

                var nameExtension = string.Join("_", new[]
                {
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }

        [Fact]
        public void ExampleSpaces()
        {
            var seeds = Enumerable.Range(0, 3);

            foreach (var seed in seeds)
            {
                var sample = Gen.Char()
                    .Select(c => char.IsControl(c) ? "<control>" : c.ToString())
                    .Advanced.SampleOneExampleSpace(seed: seed, size: 75)
                    .Take(500)
                    .Render(x => x);

                var nameExtension = string.Join("_", new[]
                {
                    $"Seed_{seed}"
                });

                Snapshot.Match(sample, SnapshotNameExtension.Create(nameExtension));
            }
        }
    }
}
