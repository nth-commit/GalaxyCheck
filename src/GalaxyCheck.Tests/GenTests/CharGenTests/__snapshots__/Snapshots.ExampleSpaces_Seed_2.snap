, (Id = 74, Distance = 29.0196...)
├> a (Id = 0, Distance = 0)
├> L (Id = 37, Distance = 14.5098...)
|  ├> b (Id = 1, Distance = 0.3922...)
|  ├> t (Id = 19, Distance = 7.4510...)
|  |  ├> c (Id = 2, Distance = 0.7843...)
|  |  ├> l (Id = 11, Distance = 4.3137...)
|  |  |  ├> d (Id = 3, Distance = 1.1765...)
|  |  |  ├> h (Id = 7, Distance = 2.7451...)
|  |  |  |  ├> e (Id = 4, Distance = 1.5686...)
|  |  |  |  └> g (Id = 6, Distance = 2.3529...)
|  |  |  |     └> f (Id = 5, Distance = 1.9608...)
|  |  |  ├> j (Id = 9, Distance = 3.5294...)
|  |  |  |  └> i (Id = 8, Distance = 3.1373...)
|  |  |  └> k (Id = 10, Distance = 3.9216...)
|  |  ├> p (Id = 15, Distance = 5.8824...)
|  |  |  ├> m (Id = 12, Distance = 4.7059...)
|  |  |  └> o (Id = 14, Distance = 5.4902...)
|  |  |     └> n (Id = 13, Distance = 5.0980...)
|  |  ├> r (Id = 17, Distance = 6.6667...)
|  |  |  └> q (Id = 16, Distance = 6.2745...)
|  |  └> s (Id = 18, Distance = 7.0588...)
|  ├> C (Id = 28, Distance = 10.9804...)
|  |  ├> u (Id = 20, Distance = 7.8431...)
|  |  ├> y (Id = 24, Distance = 9.4118...)
|  |  |  ├> v (Id = 21, Distance = 8.2353...)
|  |  |  └> x (Id = 23, Distance = 9.0196...)
|  |  |     └> w (Id = 22, Distance = 8.6275...)
|  |  ├> A (Id = 26, Distance = 10.1961...)
|  |  |  └> z (Id = 25, Distance = 9.8039...)
|  |  └> B (Id = 27, Distance = 10.5882...)
|  ├> H (Id = 33, Distance = 12.9412...)
|  |  ├> D (Id = 29, Distance = 11.3725...)
|  |  ├> F (Id = 31, Distance = 12.1569...)
|  |  |  └> E (Id = 30, Distance = 11.7647...)
|  |  └> G (Id = 32, Distance = 12.5490...)
|  ├> J (Id = 35, Distance = 13.7255...)
|  |  └> I (Id = 34, Distance = 13.3333...)
|  └> K (Id = 36, Distance = 14.1176...)
├> 3 (Id = 56, Distance = 21.9608...)
|  ├> M (Id = 38, Distance = 14.9020...)
|  ├> V (Id = 47, Distance = 18.4314...)
|  |  ├> N (Id = 39, Distance = 15.2941...)
|  |  ├> R (Id = 43, Distance = 16.8627...)
|  |  |  ├> O (Id = 40, Distance = 15.6863...)
|  |  |  └> Q (Id = 42, Distance = 16.4706...)
|  |  |     └> P (Id = 41, Distance = 16.0784...)
|  |  ├> T (Id = 45, Distance = 17.6471...)
|  |  |  └> S (Id = 44, Distance = 17.2549...)
|  |  └> U (Id = 46, Distance = 18.0392...)
|  ├>   (Id = 52, Distance = 20.3922...)
|  |  ├> W (Id = 48, Distance = 18.8235...)
|  |  ├> Y (Id = 50, Distance = 19.6078...)
|  |  |  └> X (Id = 49, Distance = 19.2157...)
|  |  └> Z (Id = 51, Distance = 20.0)
|  ├> 1 (Id = 54, Distance = 21.1765...)
|  |  └> 0 (Id = 53, Distance = 20.7843...)
|  └> 2 (Id = 55, Distance = 21.5686...)
├> # (Id = 65, Distance = 25.4902...)
|  ├> 4 (Id = 57, Distance = 22.3529...)
|  ├> 8 (Id = 61, Distance = 23.9216...)
|  |  ├> 5 (Id = 58, Distance = 22.7451...)
|  |  └> 7 (Id = 60, Distance = 23.5294...)
|  |     └> 6 (Id = 59, Distance = 23.1373...)
|  ├> ! (Id = 63, Distance = 24.7059...)
|  |  └> 9 (Id = 62, Distance = 24.3137...)
|  └> " (Id = 64, Distance = 25.0980...)
├> ( (Id = 70, Distance = 27.4510...)
|  ├> $ (Id = 66, Distance = 25.8824...)
|  ├> & (Id = 68, Distance = 26.6667...)
|  |  └> % (Id = 67, Distance = 26.2745...)
|  └> ' (Id = 69, Distance = 27.0588...)
├> * (Id = 72, Distance = 28.2353...)
|  └> ) (Id = 71, Distance = 27.8431...)
└> + (Id = 73, Distance = 28.6275...)
